src/go/k8s/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -183,6 +183,37 @@ type ClusterSpec struct {
 
 	// The name of the ServiceAccount to be used by the Redpanda pods
 	ServiceAccount *string `json:"serviceAccount,omitempty"`
+
+	// NodePools runs a StatefulSet for every NodePool. All these produce pods,
+	// that join the cluster, but are steered independently (especially replica
+	// count).
+	NodePools []NodePoolSpec `json:"nodePools,omitempty"`
+}
+
+// NodePoolSpec defines a NodePool. NodePools have their own:
+// NodeSelector, so they can be scheduled on specific cloud provider Node Pools.
+// Storage, as different NodePools may have different disk shapes.
+// Replicas, so they can be controlled independently
+// Resources, because this is tied strongly to the actual machine shape backing the NodePool.
+type NodePoolSpec struct {
+	Name string `json:"name"`
```

Comment:
I think we discussed this but I don't recall the outcome, did you consider placing a `PodTemplate` here instead of only allowing Tolerations and NodeSelector to be overridden?